packages/protocol/contracts/common/FixidityLib.sol

```diff
@@ -0,0 +1,335 @@
+    return Fraction(x * FIXED1_INT);
+  }
+
+  /**
+   * @notice Converts an int256 in the fixed point representation of this
+   * library to a non decimal. All decimal digits will be truncated.
+   */
+  function fromFixed(Fraction memory x)
+    internal
+    pure
+    returns (uint256)
+  {
+    return x.value / FIXED1_INT;
+  }
+
+  /**
+   * @notice Converts two int256 representing a fraction to fixed point units,
+   * equivalent to multiplying dividend and divisor by 10^digits().
+   * @dev
+   * Test newFixedFraction(maxFixedDiv()+1,1) fails
+   * Test newFixedFraction(1,maxFixedDiv()+1) fails
+   * Test newFixedFraction(1,0) fails
+   * Test newFixedFraction(0,1) returns 0
+   * Test newFixedFraction(1,1) returns fixed1()
+   * Test newFixedFraction(maxFixedDiv(),1) returns maxFixedDiv()*fixed1()
+   * Test newFixedFraction(1,fixed1()) returns 1
+   * Test newFixedFraction(1,fixed1()-1) returns 0
+   */
+  function newFixedFraction(
+    uint256 numerator,
+    uint256 denominator
+    )
+    internal
+    pure
+    returns (Fraction memory)
+  {
+    require(numerator <= maxNewFixed());
+    require(denominator <= maxNewFixed());
+    require(denominator != 0);
+    Fraction memory convertedNumerator = newFixed(numerator);
+    Fraction memory convertedDenominator = newFixed(denominator);
+    return divide(convertedNumerator, convertedDenominator);
+  }
+
+  /**
+   * @notice Returns the integer part of a fixed point number.
+   * @dev
+   * Test integer(0) returns 0
+   * Test integer(fixed1()) returns fixed1()
+   * Test integer(newFixed(maxNewFixed())) returns maxNewFixed()*fixed1()
+   * Test integer(-fixed1()) returns -fixed1()
+   * Test integer(newFixed(-maxNewFixed())) returns -maxNewFixed()*fixed1()
+   */
+  function integer(Fraction memory x) internal pure returns (Fraction memory) {
+    return Fraction((x.value / FIXED1_INT) * FIXED1_INT); // Can't overflow
+  }
+
+  /**
+   * @notice Returns the fractional part of a fixed point number.
+   * In the case of a negative number the fractional is also negative.
+   * @dev
+   * Test fractional(0) returns 0
+   * Test fractional(fixed1()) returns 0
+   * Test fractional(fixed1()-1) returns 10^24-1
+   * Test fractional(-fixed1()) returns 0
+   * Test fractional(-fixed1()+1) returns -10^24-1
+   */
+  function fractional(Fraction memory x) internal pure returns (Fraction memory) {
+    return Fraction(x.value - (x.value / FIXED1_INT) * FIXED1_INT); // Can't overflow
+  }
+
+  /**
+   * @notice x+y. If any operator is higher than maxFixedAdd() it
+   * might overflow.
+   * @dev
+   * Test add(maxFixedAdd(),maxFixedAdd()) returns maxUint256()-1
+   * Test add(maxFixedAdd()+1,maxFixedAdd()+1) fails
+   * Test add(maxUint256(),maxUint256()) fails
+   */
+  function add(Fraction memory x, Fraction memory y) internal pure returns (Fraction memory) {
+    uint256 z = x.value + y.value;
+    if (x.value > 0 && y.value > 0) require(z > x.value && z > y.value);
```

Comment:
Definitely can be simplified. Either by importing and using SafeMath or doing the same check inline here, rather than the excessive one that's here right now.